sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2426,6 +2426,17 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     countForMalformedJSON(0, Seq(""))
   }
 
+  test("SPARK-26745: count() for non-multiline input with empty lines") {
+    val df = spark.read.json(testFile("test-data/with-empty-line.json"))
+    val withEmptyLineData = Row(1, 2, 3) :: Row(4, 5, 6) :: Row(7, 8, 9) :: Nil
```

Comment:
Actually, can you write a test without a separate file? you could do, for instance,

```scala
withTempPath { path =>
  Seq("", """{"a":4}""", "", ...).toDF.repartition(1)
    .write.text(path.getCanonicalPath())
  assert(spark.read.json(path.getCanonicalPath()).count() === 3)
}
```